### ISSUE TITLE
Mujoco Integration

### ISSUE THREAD

**Issue by Dar-rius** [Dar-rius/zeroRL#64]
## Summary

The goal is to integrate Mujoco as the default gym in place of gymnasium.

Packages such as [mujoco](https://github.com/google-deepmind/mujoco) or [mujoco-torch](https://github.com/vmoens/mujoco-torch) should be used. If you have other alternatives, please let us know by replying to the issue. 

Before beginning the integration, propose a design concept, ensuring that it is simple and straightforward without adding multiple layers of abstraction or breaking existing abstractions and functions such as those related to Gymnasium. 

You can already draw inspiration from TorchRL [here](https://github.com/pytorch/rl/blob/main/torchrl/envs/libs/mjlab.py).

**Comment by Silanoa** [comment]
## Design concept

Before we code anything — here's what I'd do.

Keep Gymnasium. Almost everything in zeroRL already assumes it (BaseEnv, vectorize_env, SyncVectorEnv). Replacing that isn't what this issue needs. MuJoCo should just become the default for robotics.

Package-wise: use `mujoco`, move it from the optional extra into the main deps. I'd skip mujoco-torch (it depends on TorchRL and patches torch). mjlab / GPU parallel envs can wait, we don't need that yet.

Same two entry points we already have:
- strings still work: `easy_train_ppo(Ant-v5, ...)` via gym.make
- new `MujocoEnv(BaseEnv)` for custom XML — load the model, step physics, override obs/reward/done like any other BaseEnv

Small gotcha: pass a class or factory into vectorize_env, not a live instance. deepcopy and MjData don't mix well. Default obs can be qpos/qvel so smoke tests don't need a full task.

Action clipping is #66. Saving NormMeanStd is still #65.

If this looks ok I can start on a PR.

**Comment by Dar-rius** [comment]
If I well understand, you want provide a wrapper like MujocoEnv to load the simulation and if user want a environment already available in Gymansium, we use gym.make ?

**Comment by Silanoa** [comment]
Yeah that's it.

- Already in Gymnasium (Ant-v5, Humanoid-v5, …) → `gym.make` like today
- Custom XML / your own task → subclass `MujocoEnv`

Nothing else changes on the train side.

**Comment by Dar-rius** [comment]
Is possible to use a function for MujocoEnv or it manage many states ?

**Comment by Silanoa** [comment]
Both work the same way as other BaseEnvs.

`MujocoEnv` is one sim (one MjModel / MjData). For several envs you pass a class or a function into `get_env` / `vectorize_env`, same as today:

```python
get_env(lambda: MyRobot(robot.xml), num_envs=4)
```

It doesn’t batch many states inside one MujocoEnv — SyncVectorEnv still does that.

**Comment by Dar-rius** [comment]
Okay I understand, you can implement this solution and create units test to assure if all work well

**Comment by Silanoa** [comment]
Cool, I'll open a PR with MujocoEnv + unit tests.
